src/datasources/locking-api/locking-api.service.ts

```diff
@@ -25,6 +25,16 @@ export class LockingApi implements ILockingApi {
       this.configurationService.getOrThrow<string>('locking.baseUri');
   }
 
+  async getCampaignById(campaignId: string): Promise<Campaign> {
+    try {
+      const url = `${this.baseUri}/api/v1/campaigns/${campaignId}`;
+      const { data } = await this.networkService.get<Campaign>({ url });
```

Comment:
Do we need to take caching into consideration? Or does the Locking Service?

Reply:
After discussion, it has been confirmed that the Locking Service will cache. We will leave this as is for now.